com.dynamo.cr/com.dynamo.cr.bob/src/com/dynamo/bob/bundle/ZipUtil.java

```diff
@@ -46,55 +46,45 @@ private static String stripBaseDir(File baseDir, File file) {
 		return strippedPath;
 	}
 
+	private static CRC32 calculateCrc32(File file) throws IOException {
+		CRC32 crc = new CRC32();
+		InputStream fis = new FileInputStream(file);
+		byte[] bytes = new byte[16 * 1024];
+		int length;
+		while((length = fis.read(bytes)) > 0) {
+			crc.update(bytes, 0, length);
+		}
+		fis.close();
+		return crc;
+	}
+
 	private static void zipFile(ZipOutputStream zipOut, File baseDir, File file) throws IOException {
 		final String filePath = stripBaseDir(baseDir, file).replace('\\', '/');
 		final long fileSize = file.length();
 
 		ZipEntry ze = new ZipEntry(filePath);
 		ze.setSize(fileSize);
-		byte[] entryData = null;
-		CRC32 crc = null;
 
 		// Some files need to be STORED instead of DEFLATED to
 		// get "correct" memory mapping at runtime.
-		int zipMethod = ZipEntry.DEFLATED;
 		boolean isAsset = filePath.startsWith("assets");
 		if (isAsset) {
 			// Set up an uncompressed file, unfortunately need to calculate crc32 and other data for this to work.
 			// https://www.infoworld.com/article/2071337/creating-zip-and-jar-files.html
-			crc = new CRC32();
-			zipMethod = ZipEntry.STORED;
-			ze.setCompressedSize(fileSize);
-		}
-
-		ByteArrayOutputStream byteOut = new ByteArrayOutputStream();
-		try {
-			if (fileSize > 0) {
-				int count;
-				entryData = new byte[(int) fileSize];
```

Comment:
Loading the entire file into a byte array is not a great idea for really large files. It's better to read chunks and writing as we go.

Reply:
does it change something for the end user? is this step now faster/slower or something like that?

Reply:
I moved this to a utility class since we do it in a number of different places. It turns out that we should wrap the FileInputStream in a BufferedInputStream. This results in a 20% speed increase compare to the old way of reading the bytes.